fem/sidredatacollection.cpp

```diff
@@ -489,6 +504,73 @@ createMeshBlueprintTopologies(bool hasBP, const std::string& mesh_name)
 }
 
 // private method
+#ifdef MFEM_USE_MPI
+void SidreDataCollection::createMeshBlueprintAdjacencies(bool hasBP)
+{
+   ParMesh *pmesh = dynamic_cast<ParMesh*>(mesh);
+
+   const int GRP_SZ = 35;
+   char group_str[GRP_SZ];
+
+   // TODO(JRC): Separate this out into group hierarchy setup and data allocation
+   // stages like all of the other "createMeshBlueprint*" functions.
```

Comment:
I'm confused: in one place you say "... it doesn't work optimally ..." but then you say "... this lack of functionality ...". Is the case `hasBP == false` not supported right now or it is just not optimal?

In any case, it may be good to add a comment with a TODO item for the case `hasBP == false`.